lib/ignored-paths.js

```diff
@@ -201,6 +201,12 @@ class IgnoredPaths {
 
         const ig = ignore().add(DEFAULT_IGNORE_DIRS);
 
+        if (this.options.dotfiles !== true) {
+
+            // Ignore hidden folders.  (This cannot be ".*", or else it's not possible to unignore hidden files)
+            ig.add([".*/*", "!../"]);
```

Comment:
I think it could be `.*/`. It's just that you would have to unignore the folder you want to lint e.g `.foo/`. Unless the issue is with glob syntax, which I am not so familiar with.